nats.go

```diff
@@ -1542,35 +1544,70 @@ func (nc *Conn) processOpErr(err error) {
 	nc.Close()
 }
 
-// Marker to close the channel to kick out the Go routine.
-func (nc *Conn) closeAsyncFunc() asyncCB {
-	return func() {
-		nc.mu.Lock()
-		if nc.ach != nil {
-			close(nc.ach)
-			nc.ach = nil
-		}
-		nc.mu.Unlock()
-	}
+// Returns an initialized asyncCallbacksHandler object
+func newAsyncCallbacksHandler() *asyncCallbacksHandler {
```

Comment:
Seems to be only called once and very short, maybe just inline?

Reply:
actually called in a test too.